File: test-openai-sdk.py
#!/usr/bin/env python3
"""
Тестування вашого API через OpenAI Python SDK
Використання: python test-openai-sdk.py [--chat] [--stream] [--full]
"""

from openai import OpenAI
import time
import sys

# Налаштування клієнта OpenAI для вашого API
client = OpenAI(
    api_key="dummy-key",
    base_url="https://8a42c760f69d.ngrok-free.app/v1"
)

# Доступні моделі для тестування
MODELS = [
    'microsoft/phi-3-mini-4k-instruct',
    'microsoft/phi-3.5-mini-instruct', 
    'openai/gpt-4o-mini',
    'meta/llama-3.3-70b-instruct',
    'mistral-ai/mistral-small-2503'
]

def test_models():
    """Тестування різних моделей"""
    print("🚀 Тестування моделей через OpenAI Python SDK\n")
    
    try:
        # Отримуємо список моделей
        print("📋 Отримання списку моделей...")
        models_response = client.models.list()
        print(f"✅ Знайдено {len(models_response.data)} моделей\n")
        
        # Тестуємо кілька моделей
        test_prompt = "Привіт! Коротко розкажи про себе українською мовою."
        
        for model in MODELS[:3]:  # Тестуємо перші 3 моделі
            print(f"🤖 Тестування моделі: {model}")
            
            try:
                start_time = time.time()
                
                response = client.chat.completions.create(
                    model=model,
                    messages=[
                        {"role": "system", "content": "Ти корисний AI асистент, який відповідає українською мовою."},
                        {"role": "user", "content": test_prompt}
                    ],
                    temperature=0.7,
                    max_tokens=200
                )
                
                duration = int((time.time() - start_time) * 1000)
                
                print(f"✅ Відповідь ({duration}ms):")
                content = response.choices[0].message.content
                print(f'   "{content[:100]}..."')
                print(f"📊 Токени: {response.usage.total_tokens} (prompt: {response.usage.prompt_tokens}, completion: {response.usage.completion_tokens})\n")
                
            except Exception as error:
                status = getattr(error, 'status', None) or getattr(getattr(error, 'response', None), 'status', None)
                body = ''
                resp = getattr(error, 'response', None)
                if resp is not None and hasattr(resp, 'data') and resp.data is not None:
                    try:
                        import json as _json
                        body = _json.dumps(resp.data)[:500]
                    except Exception:
                        body = str(resp.data)[:500]
                msg = str(error)
                print(f"❌ Помилка{f' ({status})' if status else ''}: {body or msg}\n")
                
    except Exception as error:
        status = getattr(error, 'status', None) or getattr(getattr(error, 'response', None), 'status', None)
        body = ''
        resp = getattr(error, 'response', None)
        if resp is not None and hasattr(resp, 'data') and resp.data is not None:
            try:
                import json as _json
                body = _json.dumps(resp.data)[:500]
            except Exception:
                body = str(resp.data)[:500]
        print(f"❌ Помилка підключення{f' ({status})' if status else ''}: {body or error}")

def interactive_chat():
    """Інтерактивний чат"""
    print("\n💬 Інтерактивний чат (введіть 'exit' для виходу)\n")
    
    messages = [
        {"role": "system", "content": "Ти корисний AI асистент, який відповідає українською мовою."}
    ]
    
    model = 'microsoft/phi-3-mini-4k-instruct'  # Використовуємо стабільну модель
    
    while True:
        try:
            user_input = input("👤 Ви: ").strip()
            
            if user_input.lower() == 'exit':
                break
                
            if user_input:
                messages.append({"role": "user", "content": user_input})
                
                try:
                    print("🤖 Думаю...")
                    
                    response = client.chat.completions.create(
                        model=model,
                        messages=messages,
                        temperature=0.7,
                        max_tokens=500
                    )
                    
                    assistant_message = response.choices[0].message.content
                    messages.append({"role": "assistant", "content": assistant_message})
                    
                    print(f"🤖 AI: {assistant_message}\n")
                    
                except Exception as error:
                    print(f"❌ Помилка: {error}\n")
                    
        except KeyboardInterrupt:
            print("\n👋 До побачення!")
            break

def test_streaming():
    """Тестування стрімінгу"""
    print("\n🌊 Тестування стрімінгу...\n")
    
    try:
        response = client.chat.completions.create(
            model='microsoft/phi-3-mini-4k-instruct',
            messages=[
                {"role": "system", "content": "Ти корисний AI асистент, який відповідає українською мовою."},
                {"role": "user", "content": "Розкажи коротку історію про роботів майбутнього."}
            ],
            stream=True,
            temperature=0.7,
            max_tokens=300
        )
        
        print("🤖 AI (streaming): ")
        print("   ", end="", flush=True)
        
        for chunk in response:
            if chunk.choices[0].delta.content is not None:
                print(chunk.choices[0].delta.content, end="", flush=True)
        
        print("\n✅ Стрімінг завершено\n")
        
    except Exception as error:
        print(f"❌ Помилка стрімінгу: {error}")

def test_multiple_turns():
    """Тестування багатоходової розмови"""
    print("\n🔄 Тестування багатоходової розмови...\n")
    
    try:
        messages = [
            {"role": "system", "content": "Ти корисний AI асистент, який відповідає українською мовою."},
            {"role": "user", "content": "Як тебе звати?"},
            {"role": "assistant", "content": "Я AI асистент, створений для допомоги вам."},
            {"role": "user", "content": "А що ти можеш робити?"}
        ]
        
        response = client.chat.completions.create(
            model='microsoft/phi-3-mini-4k-instruct',
            messages=messages,
            temperature=0.7,
            max_tokens=300
        )
        
        print("🤖 Відповідь на багатоходову розмову:")
        print(f"   {response.choices[0].message.content}")
        print(f"📊 Токени: {response.usage.total_tokens}\n")
        
    except Exception as error:
        print(f"❌ Помилка: {error}")

def test_tts():
    """Інформація про TTS API"""
    print("🔊 Тестування TTS API...\n")
    
    tts_url = "https://8a42c760f69d.ngrok-free.app/tts?text=Привіт%20від%20вашого%20AI%20асистента!"
    print(f"🌐 TTS URL: {tts_url}")
    print("💡 Відкрийте це посилання в браузері, щоб прослухати аудіо\n")

def show_usage():
    """Показати інформацію про використання"""
    print("💡 Додаткові опції:")
    print("   --chat, -c     : Інтерактивний чат")
    print("   --stream, -s   : Тестування стрімінгу")
    print("   --tts, -t      : Тестування TTS")
    print("   --full, -f     : Повний тест усіх функцій")
    print("   --help, -h     : Показати цю довідку\n")

def main():
    """Головна функція"""
    args = sys.argv[1:]
    
    if '--help' in args or '-h' in args:
        show_usage()
        return
    
    if '--chat' in args or '-c' in args:
        interactive_chat()
    elif '--stream' in args or '-s' in args:
        test_streaming()
    elif '--tts' in args or '-t' in args:
        test_tts()
    elif '--full' in args or '-f' in args:
        test_models()
        test_streaming()
        test_multiple_turns()
        test_tts()
        interactive_chat()
    else:
        # Базовий тест за замовчуванням
        test_models()
        test_multiple_turns()
        show_usage()

if __name__ == "__main__":
    try:
        main()
    except KeyboardInterrupt:
        print("\n👋 Програму завершено користувачем")
    except Exception as e:
        print(f"❌ Критична помилка: {e}")

File: test-openai-sdk.mjs
#!/usr/bin/env node

/**
 * Тестування OpenAI SDK з вашим API
 * Використання: node test-openai-sdk.mjs
 */

import OpenAI from 'openai';
import readline from 'readline';

// Налаштування клієнта OpenAI для вашого API
const client = new OpenAI({
    apiKey: 'dummy-key',
    baseURL: 'https://8a42c760f69d.ngrok-free.app/v1'
});

// Доступні моделі для тестування
const MODELS = [
    'microsoft/phi-3-mini-4k-instruct',
    'microsoft/phi-3.5-mini-instruct',
    'openai/gpt-4o-mini',
    'meta/llama-3.3-70b-instruct',
    'mistral-ai/mistral-small-2503'
];

async function testModels() {
    console.log('🚀 Тестування моделей через OpenAI SDK\n');
    
    try {
        // Спочатку отримаємо список доступних моделей
        console.log('📋 Отримання списку моделей...');
        const modelsList = await client.models.list();
        console.log(`✅ Знайдено ${modelsList.data.length} моделей\n`);
        
        // Тестуємо кілька моделей
        const testPrompt = 'Привіт! Коротко розкажи про себе українською мовою.';
        
        for (const model of MODELS.slice(0, 3)) { // Тестуємо перші 3 моделі
            console.log(`🤖 Тестування моделі: ${model}`);
            
            try {
                const startTime = Date.now();
                
                const response = await client.chat.completions.create({
                    model: model,
                    messages: [
                        { role: 'system', content: 'Ти корисний AI асистент, який відповідає українською мовою.' },
                        { role: 'user', content: testPrompt }
                    ],
                    temperature: 0.7,
                    max_tokens: 200
                });
                
                const duration = Date.now() - startTime;
                
                console.log(`✅ Відповідь (${duration}ms):`);
                console.log(`   "${response.choices[0].message.content.substring(0, 100)}..."`);
                console.log(`📊 Токени: ${response.usage.total_tokens} (prompt: ${response.usage.prompt_tokens}, completion: ${response.usage.completion_tokens})\n`);
                
            } catch (error) {
                const status = error?.status || error?.response?.status;
                // Спроба дістати тіло помилки
                let detail = '';
                if (error?.response?.data) {
                    try { detail = JSON.stringify(error.response.data).slice(0, 500); } catch (_) {}
                } else if (error?.message) {
                    detail = error.message;
                }
                console.log(`❌ Помилка${status ? ` (${status})` : ''}: ${detail}\n`);
            }
        }
        
    } catch (error) {
        const status = error?.status || error?.response?.status;
        let detail = error?.message || '';
        if (error?.response?.data) {
            try { detail = JSON.stringify(error.response.data).slice(0, 500); } catch (_) {}
        }
        console.error(`❌ Помилка підключення${status ? ` (${status})` : ''}: ${detail}`);
    }
}

async function interactiveChat() {
    console.log('\n💬 Інтерактивний чат (введіть "exit" для виходу)\n');
    
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });
    
    const messages = [
        { role: 'system', content: 'Ти корисний AI асистент, який відповідає українською мовою.' }
    ];
    
    const model = 'microsoft/phi-3-mini-4k-instruct'; // Використовуємо стабільну модель
    
    const askQuestion = () => {
        rl.question('👤 Ви: ', async (input) => {
            if (input.toLowerCase() === 'exit') {
                rl.close();
                return;
            }
            
            if (input.trim()) {
                messages.push({ role: 'user', content: input });
                
                try {
                    console.log('🤖 Думаю...');
                    
                    const response = await client.chat.completions.create({
                        model: model,
                        messages: messages,
                        temperature: 0.7,
                        max_tokens: 500
                    });
                    
                    const assistantMessage = response.choices[0].message.content;
                    messages.push({ role: 'assistant', content: assistantMessage });
                    
                    console.log(`🤖 AI: ${assistantMessage}\n`);
                    
                } catch (error) {
                    console.log(`❌ Помилка: ${error.message}\n`);
                }
            }
            
            askQuestion();
        });
    };
    
    askQuestion();
}

async function testStreaming() {
    console.log('\n🌊 Тестування стрімінгу...\n');
    
    try {
        const stream = await client.chat.completions.create({
            model: 'microsoft/phi-3-mini-4k-instruct',
            messages: [
                { role: 'system', content: 'Ти корисний AI асистент, який відповідає українською мовою.' },
                { role: 'user', content: 'Розкажи коротку історію про роботів майбутнього.' }
            ],
            stream: true,
            temperature: 0.7,
            max_tokens: 300
        });
        
        console.log('🤖 AI (streaming): ');
        process.stdout.write('   ');
        
        for await (const chunk of stream) {
            const content = chunk.choices[0]?.delta?.content;
            if (content) {
                process.stdout.write(content);
            }
        }
        
        console.log('\n✅ Стрімінг завершено\n');
        
    } catch (error) {
        console.error('❌ Помилка стрімінгу:', error.message);
    }
}

async function testTTS() {
    console.log('🔊 Тестування TTS API...\n');
    
    try {
        const ttsUrl = 'https://8a42c760f69d.ngrok-free.app/tts?text=Привіт%20від%20вашого%20AI%20асистента!';
        console.log(`🌐 TTS URL: ${ttsUrl}`);
        console.log('💡 Відкрийте це посилання в браузері, щоб прослухати аудіо\n');
        
    } catch (error) {
        console.error('❌ Помилка TTS:', error.message);
    }
}

// Головна функція
async function main() {
    const args = process.argv.slice(2);
    
    if (args.includes('--chat') || args.includes('-c')) {
        await interactiveChat();
    } else if (args.includes('--stream') || args.includes('-s')) {
        await testStreaming();
    } else if (args.includes('--tts') || args.includes('-t')) {
        await testTTS();
    } else {
        // Повний тест за замовчуванням
        await testModels();
        
        if (args.includes('--full') || args.includes('-f')) {
            await testStreaming();
            await testTTS();
            await interactiveChat();
        } else {
            console.log('💡 Додаткові опції:');
            console.log('   --chat, -c     : Інтерактивний чат');
            console.log('   --stream, -s   : Тестування стрімінгу');
            console.log('   --tts, -t      : Тестування TTS');
            console.log('   --full, -f     : Повний тест усіх функцій\n');
        }
    }
}

// Запуск
main().catch(console.error);
